micrometer-core/src/main/java/io/micrometer/core/instrument/Meter.java

```diff
@@ -480,6 +481,26 @@ public Meter register(MeterRegistry registry) {
 
     }
 
+    /**
+     * Convenience interface to create new meters from tags based on a common
+     * "template"/builder. See usage in Meter implementations, e.g.: {@code Timer},
+     * {@code Counter}
+     *
+     * @param <T> Meter type
+     * @since 1.12.0
+     */
+    interface MeterProvider<T extends Meter> {
+
+        /**
+         * Registers (creates a new or gets an existing one if already exists) Meters
+         * using the provided tags.
+         * @param tags Tags to attach to the Meter about to be registered
+         * @return A new or existing Meter
+         */
+        T withTags(Iterable<? extends Tag> tags);
```

Comment:
Why `withTags()` versus the more commonly used elsewhere `tags()`?

Reply:
I don't want to speak for @jonatan-ivanov, but I think the idea was to differentiate from the `tags()` method on the builders because those return a builder and can chain, whereas this method returns a registered `Meter` and cannot be chained to add more tags. I spent some time trying to think of better naming, but I was not able to come up with something. This is definitely something we're hoping to get feedback on and improve. Ideally, we would not change the naming after RC1, but it may be worth making an exception in this case if we cannot come up with something better before merging this.

Reply:
Yeah, `tags` might be confusing since that is on the `Builder` as well as `MeterProvider`.